### PR TITLE
LASB-3027: Add the spotless gradle plugin.

### PR DESCRIPTION
**Pull request by mtac50**
## What

[Link to story](https://dsdmoj.atlassian.net/browse/LASB-3027)

- [x] Implemented Spotless gradle plugin to enforce code formatting.

## Checklist

Before you ask people to review this PR:

- [ ] Tests should be passing: `./gradlew test`
- [ ] Github should not be reporting conflicts; you should have recently run `git rebase main`.
- [ ] Avoid mixing whitespace changes with code changes in the same commit. These make diffs harder to read and conflicts more likely.
- [ ] You should have looked at the diff against main and ensured that nothing unexpected is included in your changes.
- [ ] You should have checked that the commit messages say why the change was made.

### CODE REVIEW

**Comment by mtac50** [comment]
Confirmation that code format violations cause the build pipeline to fail: https://app.circleci.com/pipelines/github/ministryofjustice/laa-crime-applications-adaptor/562/workflows/9fcbc1f9-4a4c-4344-a3df-f71ba89308db/jobs/4585

File: crime-applications-adaptor/build.gradle [hunk 1]
```diff
@@ -147,3 +148,17 @@ jsonSchema2Pojo {
 		}
 	}
 }
+
+spotless {
+	java {
+		target project.fileTree(project.rootDir) {
```
Comment (stone-dj) [inline crime-applications-adaptor/build.gradle:154]: I have a slight preference for the below with only explicit includes and no dependencies on the project attributes.

`target 'src/main/java/**/*.java', 'src/test/java/**/*.java'`

**Review by stone-dj — COMMENTED** [review]
Changes look good, just 1 piece of feedback

File: crime-applications-adaptor/build.gradle (hunk 1 above)
Comment (mtac50) [reply crime-applications-adaptor/build.gradle:154]: Great shout - I'll update!

**Pull request closed without merging**